src/screens/Signup/StepHandle.tsx

```diff
@@ -103,115 +124,212 @@ export function StepHandle() {
+                              0,
+                              state.userDomain.length * -1,
+                            ),
+                          )
+                          logger.metric('signup:handleSuggestionSelected', {
+                            method: suggestion.method,
+                          })
+                        }}
+                      />
+                    )}
+                </>
+              )}
+            {(!validCheck.handleChars || !validCheck.hyphenStartOrEnd) && (
+              <Requirement>
+                {!validCheck.hyphenStartOrEnd ? (
+                  <RequirementText>
+                    <Trans>Doesn't begin or end with a hyphen</Trans>
+                  </RequirementText>
+                ) : (
+                  <RequirementText>
+                    <Trans>Only contains letters, numbers, and hyphens</Trans>
+                  </RequirementText>
+                )}
+              </Requirement>
+            )}
+            <Requirement>
+              {(!validCheck.frontLengthNotTooLong ||
+                !validCheck.totalLength) && (
+                <RequirementText>
                   <Trans>
                     No longer than{' '}
                     <Plural
                       value={MAX_SERVICE_HANDLE_LENGTH}
                       other="# characters"
                     />
                   </Trans>
-                </Text>
-              ) : (
-                <Text style={[a.text_md, a.flex_1]}>
-                  <Trans>At least 3 characters</Trans>
-                </Text>
+                </RequirementText>
               )}
-            </View>
+            </Requirement>
           </View>
-        )}
+        </LayoutAnimationConfig>
       </View>
-      <BackNextButtons
-        isLoading={isLoading}
-        isNextDisabled={!validCheck.overall}
-        onBackPress={onBackPress}
-        onNextPress={onNextPress}
-      />
+      <Animated.View layout={native(LinearTransition)}>
+        <BackNextButtons
+          isLoading={isNextLoading}
+          isNextDisabled={!validCheck.overall || !!state.error}
+          onBackPress={onBackPress}
+          onNextPress={onNextPress}
+        />
+      </Animated.View>
     </ScreenTransition>
   )
 }
 
-function IsValidIcon({valid}: {valid: boolean}) {
+function Requirement({children}: {children: React.ReactNode}) {
+  return (
+    <Animated.View
+      style={[a.w_full]}
+      layout={native(LinearTransition)}
+      entering={native(FadeIn)}
+      exiting={native(FadeOut)}>
+      {children}
+    </Animated.View>
+  )
+}
+
+function RequirementText({children}: {children: React.ReactNode}) {
   const t = useTheme()
-  if (!valid) {
-    return <Times size="md" style={{color: t.palette.negative_500}} />
-  }
-  return <Check size="md" style={{color: t.palette.positive_700}} />
+  return (
+    <Text style={[a.text_sm, a.flex_1, {color: t.palette.negative_500}]}>
+      <Trans>{children}</Trans>
```

Comment:
```suggestion
      {children}
```
There would be no translatable text here, so I suggest removing `Trans` tags :)